extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamSegment.java

```diff
@@ -0,0 +1,69 @@
+package org.schabi.newpipe.extractor.stream;
+
+
+import javax.annotation.Nullable;
+import java.io.Serializable;
+
+public class StreamSegment implements Serializable {
+    /**
+     * Title of this segment
+     */
+    private String title;
+
+    /**
+     * Timestamp of the starting point in seconds
+     */
+    private int startTimeSeconds;
+
+    /**
+     * Direct url to this segment. This can be null if the service doesn't provide such function.
+     */
+    @Nullable
+    public String url;
+
+    /**
+     * Preview url for this segment. This can be null if the service doesn't provide such function
+     * or there is no resource found.
+     */
+    @Nullable
+    private String previewUrl = null;
+
+    public StreamSegment(String title, int startTimeSeconds) {
+        this.title = title;
+        this.startTimeSeconds = startTimeSeconds;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public void setTitle(final String title) {
+        this.title = title;
+    }
+
+    public int getStartTimeSeconds() {
+        return startTimeSeconds;
+    }
+
+    public void setStartTimeSeconds(final int startTimeSeconds) {
+        this.startTimeSeconds = startTimeSeconds;
+    }
+
+    @Nullable
+    public String getUrl() {
+        return url;
+    }
+
+    public void setUrl(@Nullable final String url) {
+        this.url = url;
+    }
+
+    @Nullable
+    public String getPreviewUrl() {
+        return previewUrl;
+    }
+
+    public void setPreviewUrl(@Nullable final String previewUrl) {
+        this.previewUrl = previewUrl.replaceAll("\\u0026", "&");
```

Comment:
this looks YouTube specific and should therefore be moved from within the function to the function call.

Reply:
Actually, it can be removed completely. The encoding when it's printed in the console is different to the one when saving the html itself from source tab in Chrome/Firefox (I checked it via `System.out.println()`) :')
So I'll wrap it with `fixThumbnailUrl(url)` then like the other `getThumbnailUrl` calls.